tests/Microsoft.Identity.Test.Unit/PublicApiTests/AdfsAcceptanceTests.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Test.Common.Core.Mocks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Identity.Test.Unit.PublicApiTests
+{
+    [TestClass]
+    public class AdfsAcceptanceTests : TestBase
+    {
+        // Possible authorities copied from: https://msazure.visualstudio.com/One/_search?action=contents&text=CanAcquireToken_UsingRefreshToken&type=code&lp=code-Project&filters=ProjectFilters%7BOne%7DRepositoryFilters%7BAzureStack-Services-Graph%7D&pageSize=25&result=DefaultCollection/One/AzureStack-Services-Graph/GBmain//src/Identity.Web.Tests/MsalTests.cs
+        [DataTestMethod]
+        [DataRow("https://localhost:3001/adfs")]
+        [DataRow("https://localhost:3001/460afc9d-718d-40c8-8d03-954")]
+        [DataRow("https://localhost:3001/contoso.int.test")]
+        public async Task AdfsAuthorityVariants_WithAdfsAuthority_Async(string authority)
+        {
+            await RunAuthCodeFlowAsync(authority, useWithAdfsAuthority: true).ConfigureAwait(false);
+        }
+
+        [DataTestMethod]
+        [DataRow("https://localhost:3001/adfs")]
+        [DataRow("https://localhost:3001/460afc9d-718d-40c8-8d03-954")]
+        [DataRow("https://localhost:3001/contoso.int.test")]
+        public async Task AdfsAuthorityVariants_WithAuthority_Async(string authority)
+        {
+            await RunAuthCodeFlowAsync(authority, useWithAdfsAuthority: false).ConfigureAwait(false);
+        }
+
+        private static async Task RunAuthCodeFlowAsync(string authority, bool useWithAdfsAuthority)
+        {
+            using (var httpManager = new MockHttpManager())
+            {
+                var builder = ConfidentialClientApplicationBuilder
+                    .Create("e68c40a5-a8e5-4250-bbea-5b43ab18cf0d");
+
+                builder = useWithAdfsAuthority ?
```

Comment:
was intention to have this take effect? seems below we call WithAdfsAuthority again regardless of branching here

Reply:
I guess it doesn't really matter anymore, since when you call .WithAuthority, internally it resolves that the authority type is ADFS and returns the adfsauthority subclass even if caller didn;t use .WithAdfsAuthority: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/6dc1a93299b1b19c2e82d6cdcaea8be455694fec/src/client/Microsoft.Identity.Client/AppConfig/AuthorityInfo.cs#L328

Reply:
`WithAuthority` has MSAL parse the authority string and tries to infer the authority. It uses the `adfs` suffix to do this.
`WithAdfsAuthority` tells MSAL that the authority is ADFS and it won't reparse the string and try to infer this.

So internally `https://localhost:3001/460afc9d-718d-40c8-8d03-954` is treated as an AAD authority when using `WithAuthority` - but the flow works because instance discovery is disabled etc. 

I would not expect `https://localhost:3001/460afc9d-718d-40c8-8d03-954` to work, but it kind of did. So I thought I'd capture this in a unit test.